src/bootstrap/download.rs

```diff
@@ -21,6 +20,12 @@ use crate::{
 
 static SHOULD_FIX_BINS_AND_DYLIBS: OnceCell<bool> = OnceCell::new();
 
+/// For this module only, don't give warnings on `try_run`, since we don't have access to a `builder` yet.
+fn try_run(config: &Config, cmd: &mut Command) -> Result<(), ()> {
+    #[allow(deprecated)]
+    config.try_run(cmd)
```

Comment:
I think this helps a little bit more, specifically when we read this from LSP/rust-analyzer on the caller stack.

```suggestion
/// `Config::try_run` wrapper for this module to avoid warnings on `try_run`, since we don't have access to a `builder` yet.
fn try_run(config: &Config, cmd: &mut Command) -> Result<(), ()> {
    #[allow(deprecated)]
    config.try_run(cmd)
```

Reply:
sure, done